gradio/pipelines_helpers.py

```diff
@@ -0,0 +1,535 @@
+"""
+Defines internal helper methods for handling transformers and diffusers pipelines.
+These are used by load_from_pipeline method in pipelines.py.
+"""
+
+from typing import Any, Dict, Optional
+
+from PIL import Image
+
+from gradio import components
+
+
+def _handle_transformers_pipeline(pipeline: Any) -> Optional[Dict[str, Any]]:
+    try:
+        import transformers
+        from transformers import pipelines
+
+    except ImportError as ie:
+        raise ImportError(
+            "transformers not installed. Please try `pip install transformers`"
+        ) from ie
+
+    def is_transformers_pipeline_type(pipeline, class_name: str):
+        cls = getattr(transformers, class_name, None)
+        return cls and isinstance(pipeline, cls)
+
+    # Handle the different pipelines. The has_attr() checks to make sure the pipeline exists in the
+    # version of the transformers library that the user has installed.
+    if is_transformers_pipeline_type(pipeline, "AudioClassificationPipeline"):
+        pipeline_info = {
+            "inputs": components.Audio(
+                sources=["microphone"],
+                type="filepath",
+                label="Input",
+                render=False,
+            ),
+            "outputs": components.Label(label="Class", render=False),
+            "preprocess": lambda i: {"inputs": i},
+            "postprocess": lambda r: {i["label"].split(", ")[0]: i["score"] for i in r},
+        }
+        return pipeline_info
+
+    elif is_transformers_pipeline_type(pipeline, "AutomaticSpeechRecognitionPipeline"):
```

Comment:
None of these need to be `elif`s since each branch returns directly.

Reply:
done